packages/flutter/lib/src/material/animated_raster.dart

```diff
@@ -0,0 +1,202 @@
+/// transitions.
+///
+/// This class depends on [Scene.toGpuImage] which is not supported on the html backend.
+///
+/// See also:
+///  * [ZoomPageTransitionsBuilder], which uses this render object to implement the fade and
+///    stretch effect efficiently.
+class _AnimatedRaster extends SingleChildRenderObjectWidget {
+  /// Create a new [_AnimatedRaster].
+  const _AnimatedRaster({
+    super.child,
+    required this.animation,
+    required this.delegate,
+  });
+
+  /// The animation that drives the [delegate].
+  final Animation<double> animation;
+
+  /// The delegate which controls how the child raster is painted.
+  final _AnimatedRasterDelegate delegate;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RenderAnimatedRaster(animation, delegate, MediaQuery.maybeOf(context)?.devicePixelRatio ?? 1.0);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RenderAnimatedRaster renderObject) {
+    renderObject
+      ..animation = animation
+      ..delegate = delegate
+      ..devicePixelRatio = MediaQuery.maybeOf(context)?.devicePixelRatio ?? 1.0;
+  }
+}
+
+/// A render object that converts its children to a [ui.Image].
+///
+/// This render object will create a GPU resident texture the first time that
+/// [_AnimatedRasterDelegate.willPaint] returns true and and provide it to
+/// [_AnimatedRasterDelegate.paintRaster]. This will be reused until `willPaint`
+/// returns false. It is up to the implementor of the raster delegate to handle
+/// painting both the rasterized and the non-rasterized child.
+///
+/// See also:
+///  * [ZoomPageTransitionsBuilder], which uses this render object to implement the fade and
+///    stretch effect efficiently.
+class _RenderAnimatedRaster extends RenderProxyBox {
+  /// Create a new [_RenderAnimatedRaster].
+  _RenderAnimatedRaster(
+    this._animation,
+    this._delegate,
+    this._devicePixelRatio,
+  );
+
+  /// The delegate that controls how the child raster is painted.
+  _AnimatedRasterDelegate get delegate => _delegate;
+  _AnimatedRasterDelegate _delegate;
+  set delegate(_AnimatedRasterDelegate value) {
+    if (value == delegate) {
+      return;
+    }
+    _delegate = value;
+    markNeedsPaint();
+  }
+
+  /// The device pixel ratio used to create the child raster.
+  double get devicePixelRatio => _devicePixelRatio;
+  double _devicePixelRatio;
+  set devicePixelRatio(double value) {
+    if (value == devicePixelRatio) {
+      return;
+    }
+    _devicePixelRatio = value;
+    markNeedsPaint();
+  }
+
+  /// The animation used to drive this render object.
+  Animation<double> get animation => _animation;
+  Animation<double> _animation;
+  set animation(Animation<double> value) {
+    if (value == animation) {
+      return;
+    }
+    animation.removeListener(markNeedsPaint);
+    _animation = value;
+    animation.addListener(markNeedsPaint);
```

Comment:
Only do this if we are attached? Otherwise you'll do it again when you get attached.